### DISCUSSION TITLE
Google oAuth broken with @supabase/auth-helpers-nextjs 0.7.1

### DISCUSSION THREAD

**Discussion by tholder** [supabase/supabase#14760] [Questions] [answered]
Upgraded from auth-helpers-nextjs 0.6.0 to 0.71.

This required a change in creation of server client as shown:

![CleanShot 2023-06-02 at 10 18 31](https://github.com/supabase/supabase/assets/41235/d47aa2ff-94a8-484d-b717-e6e37f42efce)

The result is that Google oAuth is now broken with code_challenge_method s256 error.

![CleanShot 2023-06-02 at 10 01 59](https://github.com/supabase/supabase/assets/41235/deae1e19-f0f9-497a-849d-040bc5a99355)

Is there some additional change required? I've not changed any of my auth logic.

**Comment by estebangarcia21** [comment]
From my other discussion #14749

>To get this working locally, I had to upgrade my Supabase CLI to the latest version and restart the service by doing supabase stop and supabase start. You can find out how to upgrade at https://supabase.com/docs/guides/cli.

Try this out if you are running Supabase locally for development.

**Reply by tholder** [reply]
Thanks, tried that but no luck. Issue persists.

**Comment by silentworks** [comment, marked as answer]
Yes you need to check the docs for the auth helpers. We are now using the Proof Key for Code Exchange (PKCE) flow for better server side authentictation. This means you can now do full auth via the server and not have the client involved at all like before. We have covered this in the docs that you need to have a callback route to handle the challenge code.

https://supabase.com/docs/guides/auth/auth-helpers/nextjs#migrating-to-v07x

**Comment by J0** [comment]
Going to mark this as closed now since it seems like the issue may not be generally relevant at this time. Let us know if there are further issues
